src/client/measurementProtocolMapper.ts

```diff
@@ -137,6 +137,18 @@ const convertImpressionToMeasurementProtocol = (
     }, {});
 };
 
-export const measurementProtocolKeysMappingValues: String[] = Object.keys(measurementProtocolKeysMapping).map(key => Object(measurementProtocolKeysMapping)[key]);
-export const productKeysMappingValues = Object.keys(productKeysMapping).map(key => Object(productKeysMapping)[key]);
-export const impressionKeysMappingValues = Object.keys(impressionKeysMapping).map(key => Object(impressionKeysMapping)[key]);
\ No newline at end of file
+const measurementProtocolKeysMappingValues = keysOf(measurementProtocolKeysMapping).map(key => Object(measurementProtocolKeysMapping)[key]);
+const productKeysMappingValues = keysOf(productKeysMapping).map(key => Object(productKeysMapping)[key]);
+const impressionKeysMappingValues = keysOf(impressionKeysMapping).map(key => Object(impressionKeysMapping)[key]);
+
+const productSubKeysMatchGroup = productKeysMappingValues.join("|")
+const impressSubKeysMatchGroup = impressionKeysMappingValues.join("|")
+const productKeyRegex = new RegExp(`^(pr[0-9]+)(${productSubKeysMatchGroup})$`)
+const impressionKeyRegex = new RegExp(`^((il[0-9]+pi[0-9]+)(${impressSubKeysMatchGroup}))|(il[0-9]+nm)$`)
+
+export const isMeasurementPrototocolKey = (key: string): boolean => {
+    if (productKeyRegex.test(key) || impressionKeyRegex.test(key) || measurementProtocolKeysMappingValues.indexOf(key) != -1) {
+        return true;
+    }
+    return false;
+}
```

Comment:
```suggestion
export const isMeasurementPrototocolKey = (key: string): boolean => {
    return productKeyRegex.test(key) || impressionKeyRegex.test(key) || measurementProtocolKeysMappingValues.indexOf(key) != -1);
}
```

small detail: unnecessary if 😄